modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/PreviewJournalArticleContentDashboardItemVersionAction.java

```diff
@@ -0,0 +1,159 @@
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.HttpComponentsUtil;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.Locale;
+
+import javax.portlet.PortletResponse;
+import javax.portlet.PortletURL;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Cristina González
+ */
+public class PreviewJournalArticleContentDashboardItemVersionAction
+	implements ContentDashboardItemVersionAction {
+
+	public PreviewJournalArticleContentDashboardItemVersionAction(
+		AssetDisplayPageFriendlyURLProvider assetDisplayPageFriendlyURLProvider,
+		HttpServletRequest httpServletRequest, JournalArticle journalArticle,
+		Language language,
+		LayoutDisplayPageProviderTracker layoutDisplayPageProviderTracker,
+		LayoutLocalService layoutLocalService,
+		LayoutSEOLinkManager layoutSEOLinkManager, Portal portal,
+		RequestBackedPortletURLFactory requestBackedPortletURLFactory) {
+
+		_assetDisplayPageFriendlyURLProvider =
+			assetDisplayPageFriendlyURLProvider;
+		_httpServletRequest = httpServletRequest;
+		_journalArticle = journalArticle;
+		_language = language;
+		_layoutDisplayPageProviderTracker = layoutDisplayPageProviderTracker;
+		_layoutLocalService = layoutLocalService;
+		_layoutSEOLinkManager = layoutSEOLinkManager;
+		_portal = portal;
+		_requestBackedPortletURLFactory = requestBackedPortletURLFactory;
+	}
+
+	@Override
+	public String getIcon() {
+		return "view";
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		if (_journalArticle.isDraft()) {
+			return "preview-draft";
+		}
+
+		return _language.get(locale, "preview");
+	}
+
+	@Override
+	public String getName() {
+		return "preview";
+	}
+
+	public Type getType() {
+		return Type.BLANK;
+	}
+
+	@Override
+	public String getURL() {
+		ViewJournalArticleContentDashboardItemAction
+			viewJournalArticleContentDashboardItemAction =
+				new ViewJournalArticleContentDashboardItemAction(
+					_assetDisplayPageFriendlyURLProvider, _httpServletRequest,
+					_journalArticle, _language,
+					_layoutDisplayPageProviderTracker, _layoutLocalService,
+					_layoutSEOLinkManager, _portal);
+
+		String displayPageTemplateURL =
+			viewJournalArticleContentDashboardItemAction.getURL();
+
+		if (Validator.isNotNull(displayPageTemplateURL)) {
+			displayPageTemplateURL = HttpComponentsUtil.addParameter(
+				displayPageTemplateURL, "p_l_mode", Constants.PREVIEW);
+
+			return HttpComponentsUtil.addParameter(
+				displayPageTemplateURL, "version",
+				_journalArticle.getVersion());
+		}
+
+		if (Validator.isNull(_journalArticle.getDDMTemplateKey())) {
+			return StringPool.BLANK;
```

Comment:
I do not know if it is ok to return a blank URL here , maybe  it is better to add this logic to the isShow method in PreviewJournalArticleContentDashboardItemVersionActionProvider class.

Reply:
We do this with all the Actions :(. What do you think it is a good approach?

Reply:
Maybe we should not send to front an action with a blank URL because we know that this action should not show, so for me send an action with a blank URL knowing that it should not show is not a good approach. This is only my opinion :)

Reply:
We don't need to send it to fronted, we filtered in the provider, do you think this is Ok?

Reply:
Yes, it is ok, I would add this in the provider to the case when the content does not have display page template and does not have template.

Reply:
Hey @Mikellorza !!

Sorry I just understand the problem you were talking about, the filter we use in the tracker, only filter null URLs, not blank, I have changed it to null, and now the URLs are not returned to the frontend, could you take a look?